src/test/scala/com/feedzai/cosytest/CleanUp.scala

```diff
@@ -20,22 +18,26 @@ trait CleanUp extends TestSuite with BeforeAndAfterAll {
   private val WorkingDir = Paths.get("").toAbsolutePath
   private val setup = DockerComposeSetup("Dummy", Seq.empty, WorkingDir, Map.empty)
 
+  def dockerSetups: Seq[DockerComposeSetup]
+
   protected abstract override def afterAll(): Unit = {
     super.afterAll()
 
     // Fetch containers ids if present
-    val command = Seq("docker", "ps", "-aq")
-    val fetchedIds = setup.runCmdWithOutput(command, WorkingDir.toFile, Map.empty, 10.seconds)
-
-    val removedContainers = fetchedIds match {
-      case Success(containers) => stopAllContainers(containers) && removeAllContainers(containers)
-      case Failure(f) =>
-        logger.warn("Failed to retrieve containers ids...", f)
-        false
+    val containerIds = dockerSetups.flatMap(setup => setup.getProjectContainerIds())
```

Comment:
What do you think about having a `cleanUp` method in the setup itself, that cleans all the associated resources? Here you would just say `dockerSetups.foreach(_.cleanUp())`.

Reply:
I like that option